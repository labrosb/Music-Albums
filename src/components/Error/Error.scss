@import '../../styles/colors.scss';
@import '../../styles/mixins.scss';

.error-content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 320px;
  margin: 8% auto auto auto;
  padding: 5px 10px 20px 10px;
  background-color: color(secondary);
  border-radius: 12px;
  .error-title {
    // med-label mixin
    @include med-label(primary-contrast, center);
  }
  .error-text {
    // text mixin
    @include text(normal, null);
  }
}
