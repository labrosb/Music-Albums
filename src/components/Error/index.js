import React from 'react';
import PropTypes from 'prop-types';
import Header from '../Common/Header';
import './Error.scss';

function Error({ header, message, active, noBackButton }) {
  return (
    <>
      <Header title={header} {...{ active, noBackButton }} />
      <div className="error-content">
        <h2 className="error-title">Error!</h2>
        <p className="error-text">{message || 'An error occured!'}</p>
      </div>
    </>
  );
}

export default Error;

Error.propTypes = {
  header: PropTypes.string,
  message: PropTypes.string,
  active: PropTypes.string,
  noBackButton: PropTypes.bool
};
