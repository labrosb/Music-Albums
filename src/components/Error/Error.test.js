/* eslint-disable no-undef */
import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Error from './index';

configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Error header="My Header" message="Error message" />);
});

describe('Error page component', () => {
  it('Should render error message passed via props correctly', () => {
    const errorMessage = wrapper.find('.error-text').text();

    expect(errorMessage).toEqual('Error message');
  });

  it('Should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
