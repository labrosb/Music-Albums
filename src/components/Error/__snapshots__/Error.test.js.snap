// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page component Matches snapshot 1`] = `
<Fragment>
  <Header
    title="My Header"
  />
  <div
    className="error-content"
  >
    <h2
      className="error-title"
    >
      Error!
    </h2>
    <p
      className="error-text"
    >
      Error message
    </p>
  </div>
</Fragment>
`;

exports[`Error page component Should Mmatch snapshot 1`] = `
<Fragment>
  <Header
    title="My Header"
  />
  <div
    className="error-content"
  >
    <h2
      className="error-title"
    >
      Error!
    </h2>
    <p
      className="error-text"
    >
      Error message
    </p>
  </div>
</Fragment>
`;

exports[`Error page component Should match snapshot 1`] = `
<Fragment>
  <Header
    title="My Header"
  />
  <div
    className="error-content"
  >
    <h2
      className="error-title"
    >
      Error!
    </h2>
    <p
      className="error-text"
    >
      Error message
    </p>
  </div>
</Fragment>
`;
